`@plone/components` storybook reorganization, better docs (#6940)
plone/volto: packages/components/src/stories/Introduction.mdx

# `@plone/components`

> This package contains ReactJS basic components for Plone 7 and Plone's frontend modular architecture.

Reviewer: It looks like they dropped "JS" from their brand name: https://react.dev/

```suggestion
This package contains basic React components for Plone 7 and Plone's frontend modular architecture.
```

> The purpose of this package is to provide an agnostic set of baseline components to build Plone sites upon.

Reviewer: ```suggestion
The purpose of this package is to provide an agnostic set of baseline components upon which to build Plone sites.
```

> `@plone/components` is based on [React Aria Components](https://react-spectrum.adobe.com/react-aria/components.html), the documentation there applies also to all the components in this package.

Reviewer: ```suggestion
`@plone/components` is based on [React Aria Components](https://react-spectrum.adobe.com/react-aria/components.html).
Its documentation applies to all the components in this package.
```

## Design decisions and assumptions

This package provides a set of simple, ready to use components with the following features:

- Agnostic (not tied to any known design system)
- Composable (they are ready to be the building blocks of other complex components)
- Theme-able (they can be themed, using both the provided basic theme as a baseline and custom CSS properties)
- Data-driven-less (they are dumb, presentational components)
- i18n-less (they do not provide i18n machinery or attached to any i18n framework or library)
- Built on a renowned headless CMS components library: [React Aria Components (RAC)](https://react-spectrum.adobe.com/react-aria/components.html)

> Since we are using RAC as the base, the styling of this package components are done via the [RAC styling options](https://react-spectrum.adobe.com/react-aria/styling.html).

Reviewer: ```suggestion
Since we are using RAC as the base, the styling of these package components is done via the [RAC styling options](https://react-spectrum.adobe.com/react-aria/styling.html).
```

See the `react-aria-components` section below for more information about RAC.
This package provides a simple and basic set of CSS rules to build upon.
Alternatively, you can bring your own styles, removing the basic styling or complementing it, and build on top of it.

> You can even use RAC to use other CSS utility libraries, like TailwindCSS.

Reviewer: There's a space in the brand name.

```suggestion
You can even use RAC to use other CSS utility libraries, like Tailwind CSS.
```

## Styling

This package provides two type of components.

> One using TailwindCSS and the other using pure CSS.

Reviewer: ```suggestion
One uses Tailwind CSS, and the other uses pure CSS.
```

> The TailwindCSS components are built using the TailwindCSS utility classes.

Reviewer: ```suggestion
The Tailwind CSS components are built using the Tailwind CSS utility classes.
```

The pure CSS components are built using the CSS rules provided in the package.

## Tailwind

> This package provides a set of components that use TailwindCSS utility classes.

Reviewer: ```suggestion
This package provides a set of components that use Tailwind CSS utility classes.
```
